ftw/zipextract/zipextracter.py

```diff
@@ -227,15 +227,11 @@ def folder_exists(path):
 
         portal = api.portal.get()
 
-        try:
-            folder = portal.unrestrictedTraverse(path)
-        except AttributeError:
-            return False
-        except KeyError:
-            return False
-        return (IFolderish.providedBy(folder)
-                and '/'.join(folder.getPhysicalPath()) == os.path.normpath(path)
-                and folder)
+        folder = portal.unrestrictedTraverse(path, default=False)
+        if (folder
+                and IFolderish.providedBy(folder)
+                and '/'.join(folder.getPhysicalPath()) == os.path.normpath(path)):
+            return folder
```

Comment:
You give nothing back when it's false. I would prefer this:
```suggestion
        return (folder
                and IFolderish.providedBy(folder)
                and '/'.join(folder.getPhysicalPath()) == os.path.normpath(path))

```

Reply:
No return means returns `None` and `None` is falesish so it doesn't really make a difference. But I can readd the `return False` if you really want it XD.

Reply:
If someone uses this function like `folder_exists(bla)==False` then it's not the same anymore, so it's a breaking change

Reply:
Also the suggested change does not do the same as my code. I return the folder object or None. Whereas your code returns `True` or `False`.

Reply:
TBH the naming of this method is just bad. By the name you'd expect a `True` or `False` but even before it did not do that. The previous code is a bit confusing as well because normally you'd expect that `return (True and True and 'hello')` would return `True` because every argument in there is Trueish. But haha! Computer says no it will return `'hello'` instead. So even before it returned either `False` or the Plone object.

Reply:
Ah yes, you're right. I overlooked that the folder is returned because of the naming.